common/common.h

```diff
@@ -277,6 +278,66 @@ struct gpt_params {
     std::string lora_outfile = "ggml-lora-merged-f16.gguf";
 };
 
+enum llama_example {
+    LLAMA_EXAMPLE_ALL,
+    LLAMA_EXAMPLE_SERVER,
+    LLAMA_EXAMPLE_MAIN,
+};
+
+struct llama_arg {
+    std::set<enum llama_example> examples = {LLAMA_EXAMPLE_ALL};
+    std::vector<std::string> args;
+    std::string value_ex;
+    std::string env;
+    std::string help;
+    std::function<bool(void)>        handler_void   = nullptr;
+    std::function<bool(std::string)> handler_string = nullptr;
+    std::function<bool(bool)>        handler_bool   = nullptr;
+    std::function<bool(int)>         handler_int    = nullptr;
+    std::function<bool(float)>       handler_float  = nullptr;
+
+    llama_arg(std::vector<std::string> args, std::string help, std::function<bool(std::string)> handler) : args(args), help(help), handler_string(handler) {}
+
+    llama_arg(std::vector<std::string> args, std::string help, std::function<bool(bool)> handler) : args(args), help(help), handler_bool(handler) {}
+
+    llama_arg(std::vector<std::string> args, std::string help, std::function<bool(void)> handler) : args(args), help(help), handler_void(handler) {}
+
+    llama_arg & set_examples(std::set<enum llama_example> _examples) {
+        examples = std::move(_examples);
+        return *this;
+    }
```

Comment:
Avoid identifiers with leading underscore:

```suggestion
    llama_arg & set_examples(std::set<enum llama_example> examples) {
        this->examples = std::move(examples);
        return *this;
    }
```